app/src/main/java/com/example/recyclerview/MainActivity.java

```diff
@@ -20,51 +26,80 @@
 import retrofit2.Response;
 
 
+
+
 public class MainActivity extends AppCompatActivity {
 
     private static final String TAG = MainActivity.class.getSimpleName();
     private RecyclerView mRecyclerView;
     private RecyclerView.Adapter mAdapter;
     private RecyclerView.LayoutManager mLayoutManager;
     private ArrayList<FeedItem> mArticleList = new ArrayList<>();
-
+    private ArrayList<Source> mSourcesList = new ArrayList<>();
+    SharedPreferences prefs;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         setTitle("News");
-        mRecyclerView = (RecyclerView) findViewById(R.id.my_recycler_view);
+        mRecyclerView = findViewById(R.id.my_recycler_view);
+        prefs = getSharedPreferences("SourcesAllowance", MODE_PRIVATE);
 
         mLayoutManager = new LinearLayoutManager(this);
         mRecyclerView.setLayoutManager(mLayoutManager);
 
-        mAdapter = new MyAdapter(mArticleList);
+        mAdapter = new FeedItemAdapter(mArticleList);
         mRecyclerView.setAdapter(mAdapter);
 
-        startNetworkRequest("the-next-web");
-        startNetworkRequest("techcrunch");
-        startNetworkRequest("al-jazeera-english");
+        startNetworkRequest_sources("en");
```

Comment:
"en" je předpokládám nějaká defaultní hodnota, aby tam bylo aspoň něco. Bylo by fajn tam mít komentář nebo to pojmenovat jako konstantu  private static final String DEFAULT_SOURCES